app/.gitlab-ci.yml

```diff
@@ -279,6 +279,21 @@ test:frontend:
       changes:
       - app/**/*
 
+test:frontend-prettier:
+  needs: ["build:frontend"]
+  stage: test
+  image: $FRONTEND_TAG
+  inherit:
+    # no docker login
+    default: false
+  script:
+    - cd /app
+    - npx prettier --check src/*
+  rules:
+    - if: ($DO_CHECKS == "true") && ($CI_COMMIT_BRANCH == $RELEASE_BRANCH)
+      changes:
+      - app/frontend/**/*
```

Comment:
This should probably be

```yaml
  rules:
    - if: ($DO_CHECKS == "true") && ($CI_COMMIT_BRANCH == $RELEASE_BRANCH)
      changes:
      - app/**/*
    - if: ($DO_CHECKS == "true") && ($CI_COMMIT_BRANCH != $RELEASE_BRANCH)
      changes:
      - app/proto/**/*
      - app/frontend/**/*
```

(Also `test:frontend` should probs be this too)

Reply:
> ```yaml
>   rules:
>     - if: ($DO_CHECKS == "true") && ($CI_COMMIT_BRANCH == $RELEASE_BRANCH)
>       changes:
>       - app/**/*
>     - if: ($DO_CHECKS == "true") && ($CI_COMMIT_BRANCH != $RELEASE_BRANCH)
>       changes:
>       - app/proto/**/*
>       - app/frontend/**/*
> ```

@aapeliv thanks for looking, I was indeed unsure about which rules should apply here.
Some questions to help me understand what is going on:

does this mean that for a
- commit on release branch (develop?)
  => run prettier check as soon as there is a change to a file that matches app/**/*
- commit not on release branch (PR branches, etc, ...)
 => run prettier check as soon as there is a change to a file that matches either app/proto/ or app/frontend/

If so, why do we need to run prettier on a release branch even if nothing changed to files that should have been formatted with prettier. Is it just to be sure because it is the release branch?

Why include proto folder for non release branches? I don't thinks these changes could affect formatting of frontend. But that's just my first idea.

I do think including proto folder for tests does make sense.

Reply:
I've updated the rules as you suggested for test:frontend.
For test:frontend-prettier, I've done the same except that I left out the `protos` folder. Let me know if that line would still be required for this PR.